abaplint/abaplint-sci-server
# Docker improvements

Closes: #96, Closes: #253, Closes: #256 

OK, multiple changes here:
1) renames in package.json :) There was still cloudforge name
2) move docker files back to docker dir
3) better scripts in bin, including new `test` and `cleanup` and reuse them in GA flow
4) Better structure in GA flow, use version 2 of `docker/build-push-action`
  - test-js no meaningful changes
  - test-docker - just tests the build but no publish
  - publish-docker - dependent on success of test-docker, and `if` used just once on job level
  - run only on commit to master or PR to master (before it was for any commit anywhere)
5) `index.ts` in the root => more consistent addressing from npm scripts and docker file
6) Some docs on the new features. Ans also "CI check list" section as a reminder what to be checked if the CI flow is adjusted

CI flow tested on https://hub.docker.com/repository/docker/sbcgua/abaplint-backend-dev but leaves some room for potential mistake because of the refs change in last commit. But hopefully should work. See also https://github.com/sbcgua/abaplint-sci-server/runs/1205277789?check_suite_focus=true
